### PR TITLE
[http-spec] add link case of server driven pagination test

### PR DESCRIPTION
**Pull request by tadelesh**
only cover next link scenario for pagination. more complex case could be added later.

### CODE REVIEW

**Review by weidongxu-microsoft — COMMENTED** [review]
LGTM for the scenario.

Unsure the location. For azure it is https://github.com/Azure/typespec-azure/tree/main/packages/azure-http-specs/specs/payload

**Comment by msyyc** [comment]
> LGTM for the scenario.
> 
> Unsure the location. For azure it is https://github.com/Azure/typespec-azure/tree/main/packages/azure-http-specs/specs/payload/pageable

I think we shall try to make the position same as much as possible between these 2 specs. @tadelesh

**Comment by tadelesh** [comment]
i think current azure test under `specs/payload/pageable` is duplicated with `specs/azure/core/page`. shall we remove the formal one?

**Comment by msyyc** [comment]
`specs/azure/core/page` only for azure and `specs/payload/pageable` could be for azure or unbranded I understand.

**Comment by weidongxu-microsoft** [comment]
> i think current azure test under `specs/payload/pageable` is duplicated with `specs/azure/core/page`. shall we remove the formal one?

I think it is by Pan.

I'm always happy to remove that list with a request body.

**Comment by tadelesh** [comment]
sync with weidong. i moved tests to payload and weidong will move the azure test to azure folder.

File: packages/http-specs/specs/payload/pageable/main.tsp [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+    ```json
+    { "pets": [
+        { "id": "1", "name": "dog" },
+        { "id": "2", "name": "cat" }
+      ],
+      "links": {
+        "next": "/payload/pageable/server-driven-pagination/link/nextPage"
+      }
+    }
+    2. Next page request:
```
Comment (weidongxu-microsoft) [inline packages/http-specs/specs/payload/pageable/main.tsp:37]:
```

**Review by weidongxu-microsoft — APPROVED** [review]

**Pull request merged**